### PR TITLE
ddl: support rename the same table name

### PR DESCRIPTION
**Pull request by winkyao**
To be consistent with mysql:

```
mysql> alter table test.not_exists rename test.not_exists;
ERROR 1146 (42S02): Table 'test.not_exists' doesn't exist
mysql> alter table test.t1 rename test.t1;
Query OK, 0 rows affected (0.00 sec)
```

### CODE REVIEW

File: ddl/ddl_db_test.go [hunk 1]
```diff
@@ -1552,12 +1553,19 @@ func (s *testDBSuite) testRenameTable(c *C, sql string) {
...
 	failSQL = fmt.Sprintf(sql, "test1.t2", "test_not_exist.t")
 	s.testErrorCode(c, failSQL, tmysql.ErrErrorOnRename)
-	failSQL = fmt.Sprintf(sql, "test1.t2", "test1.t2")
-	s.testErrorCode(c, failSQL, tmysql.ErrTableExists)
+
+	// for the same table name
+	s.tk.MustExec("use test1")
+	s.tk.MustExec("create table if not exists t (c1 int, c2 int)")
+	s.tk.MustExec("create table if not exists t1 (c1 int, c2 int)")
+	s.tk.MustExec(fmt.Sprintf(sql, "test1.t", "test1.t"))
```
Comment (zimulala) [inline ddl/ddl_db_test.go]: It seems this test is the same as next test. Could we update it to `fmt.Sprintf(sql, "test1.t", "t"`?

**Comment by jackysp** [comment]
LGTM

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
@zimulala we need three LGTMs to approve this PR, PTAL again.

**Comment by shenli** [comment]
If the following statements are running concurrently:
```
create  table test.not_exists (c int);
alter table test.not_exists rename test.not_exists;
```
What will happen?

**Comment by winkyao** [comment]
@shenli Nice catch, I'm working on it.

**Comment by winkyao** [comment]
@shenli Actually, If the following statements are running concurrently:

```
create  table test.not_exists (c int);
alter table test.not_exists rename test.not_exists;
```

if RenameTable run before createTable finished,  `alter table test.not_exists rename test.not_exists` will report table not exists error, if RenameTable run after createTable, it will do nothing. In onRenamTable, we will check oldTable exists and newTable not exists, it will be handled expectedly.

**Comment by zimulala** [comment]
If we change the second SQL as follows:
```
create  table test.exists (c int);
alter table test.not_exists rename test.exists;
```
There are running concurrently. If RenameTable run after createTable, I think it will call `t.DropTable` and `t.CreateTable`.

**Comment by winkyao** [comment]
@zimulala It's another issue, this PR fix the same name. you can report a new issue for this case.

**Comment by shenli** [comment]
LGTM

**Pull request merged**